pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        array = (c_double * size.value)()
+        status = self._dll.getIdealPoleZerosResponse(array, size.value, column.value)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        values = [float(val) for val in array]
+        return values
+
+    def transfer_function_response(self):
+        """Return ideal filter transfer function parameters.
+
+        Returns
+        -------
+        str
+            The requested parameter array.
+
+
+            str
+                Multi-line string where each line contains a coefficient from
+                the numerator and/or the denominator of the transfer function.
+                The coefficient for the highest-order term is first and the terms are in decreasing order.
+        """
+        size = c_int()
+        status = self._dll.getIdealTransferFunctionResponseSize(byref(size))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        transfer_function_response_string = self._dll_interface.get_string(
+            self._dll.getIdealTransferFunctionResponse, max_size=size.value
+        )
+        return transfer_function_response_string
+
+    @property
+    def vsg_analysis_enabled(self) -> bool:
+        """Flag indicating if the offset due to source resistor in frequency and time responses is enabled.
+
+        Returns
+        -------
+        bool
+        """
+
+        vsg_analysis_enabled = c_bool()
+        status = self._dll.getVSGAnalsyis(byref(vsg_analysis_enabled))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        return bool(vsg_analysis_enabled.value)
+
+    @vsg_analysis_enabled.setter
+    def vsg_analysis_enabled(self, filter_vsg_analysis_enabled: bool):
+        status = self._dll.setVSGAnalsyis(filter_vsg_analysis_enabled)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def frequency_response(
+        self,
+        y_axis_parameter=FrequencyResponseColumn.MAGNITUDE_DB,
+        minimum_frequency="200 MHz",
+        maximum_frequency="5 GHz",
+        vsg_analysis_enabled=False,
+    ):
+        """Return the ideal filter frequency response for the given parameters.
+
+        Parameters
+        ----------
+        y_axis_parameter: `FrequencyResponseColumn`, optional
+            Parameter to return. The default is frequency response magnitude in dB.
+        minimum_frequency: str, optional
+            The default is ``200 MHz``.
+        maximum_frequency: str, optional
+            The default is ``5 GHz``.
+        vsg_analysis_enabled: bool, optional
+            The default is ``False``.
+
+        Returns
+        -------
+        tuple: The tuple contains
+                list of str:
+                    The defined frequency range.
+                list of str:
+                    The requested parameter.
```

Comment:
```suggestion
        tuple
            The tuple contains two lists of strings. The first is a list
            of the defined frequency ranges, and the second is a
            list of the requested parameters.
```
Is there a way to write summarize the data format for the tuple in the first line? I don't think what was there will work because Sphinx won't know how to format the text.